src/main/java/org/jabref/logic/texparser/DefaultTexParser.java

```diff
@@ -115,19 +117,14 @@ private void matchCitation(Path file, int lineNumber, String line) {
      */
     private void matchNestedFile(Path file, List<Path> texFiles, List<Path> referencedFiles, String line) {
         Matcher includeMatch = INCLUDE_PATTERN.matcher(line);
-        StringBuilder include;
 
         while (includeMatch.find()) {
-            include = new StringBuilder(includeMatch.group(INCLUDE_GROUP));
-
-            if (!include.toString().endsWith(TEX_EXT)) {
-                include.append(TEX_EXT);
-            }
+            String include = includeMatch.group(INCLUDE_GROUP);
 
-            Path folder = file.getParent();
-            Path inputFile = (folder == null)
-                    ? Paths.get(include.toString())
-                    : folder.resolve(include.toString());
+            Path inputFile = file.getParent().resolve(
+                    include.endsWith(TEX_EXT)
+                            ? include
+                            : String.format("%s%s", include, TEX_EXT));
 
             if (!texFiles.contains(inputFile)) {
```

Comment:
I'd suggest to also check if the file exists in the file system before adding it here.